tests/python/tank_test/tank_test_base.py

```diff
@@ -333,40 +407,46 @@ def setUp(self, parameters=None):
                              "mac_path": self.pipeline_config_root,
                              "linux_path": self.pipeline_config_root}
 
-
-
         # add files needed by the pipeline config
         pc_yml = os.path.join(self.pipeline_config_root, "config", "core", "pipeline_configuration.yml")
         pc_yml_data = ("{ project_name: %s, use_shotgun_path_cache: true, pc_id: %d, "
                        "project_id: %d, pc_name: %s}\n\n" % (self.project["tank_name"],
                                                              self.sg_pc_entity["id"],
                                                              self.project["id"],
                                                              self.sg_pc_entity["code"]))
-        self.create_file(pc_yml, pc_yml_data)
+        if self._do_io:
```

Comment:
I've considered going the inheritance route, and it would probably work, but I didn't. One of the reasons is that I was worried that I might be breaking other repos using TankTestBase already. By making my change additive to the class instead of substractive by turning some code in a base class, there's less chance I've broken something.

I'll agree though that it's bad design. :(